tests/test_core.py

```diff
@@ -4741,6 +4741,38 @@ def test_unicode_js_library(self):
     self.emcc_args += ['--js-library', path_from_root('tests', 'unicode_library.js')]
     self.do_run(open(os.path.join(self.get_dir(), 'main.cpp'), 'r').read(), u'Unicode snowman \u2603 says hello!')
 
+  def test_js_lib_dep_memset(self):
+    open('lib.js', 'w').write(r'''
+mergeInto(LibraryManager.library, {
+  depper__deps: ['memset'],
```

Comment:
It's still the case that `deps_info.json` is the only way for a JS library to depend on something in a C system library.

About the second paragraph, if the user defined something on the C side, it wouldn't be needed as part of `__deps` anyhow - the deps are for system library code, it determines which system libs we link in. If myFunctionOnCSide is in one of the source or bitcode files the users gives as input to emcc, they will be linked in anyhow.

(Although the user may need to add KEEPALIVE if there is no other use of the function - in theory it might be nice if the deps handled that somehow, but it might also be more confusing, I'm not sure.)